server/api/school/school.controller.js

```diff
@@ -113,6 +114,40 @@ exports.destroy = function(req, res) {
   });
 };
 
+function archiveStudents(id) {
+  return new Promise(function(resolve, reject) {
```

Comment:
You don't need to wrap this in a Promise since mongoose Model.update returns queries that are promise like interface.

This is fine for now. I'll refactor this when I merge the other changes.